### PR TITLE
feature: Add signup marketing email checkbox

### PR DESCRIPTION
**Pull request by olerichter00**
The type of this PR is: **Feature**

This PR resolves [CX-1419]

### Description

| Simulator Screen Shot - iPhone 8 - 2021-07-30 at 13 21 07 |
||

### PR Checklist (tick all before merging)

- [x] I have included screenshots or videos to illustrate my changes, or I have not changed anything that impacts the UI.
- [x] I have tested my changes on **iOS** and **Android**.
- [x] I have added tests for my changes, or my changes don't require testing, or I have included a link to a separate Jira ticket covering the tests.
- [x] I have added a feature flag, or my changes don't require a feature flag. ([How do I add one?](https://github.com/artsy/eigen/blob/master/docs/developing_a_feature.md))
- [x] I have documented any follow-up work that this PR will require, or it does not require any.
- [x] I have added an app state migration, or my changes do not require one. ([What are migrations?](https://github.com/artsy/eigen/blob/master/docs/adding_state_migrations.md))
- [x] I have added a changelog entry below or my changes do not require one.

<details><summary>Changelog updates</summary>

#### Cross-platform user-facing changes

- Add separate marketing email checkbox to new signup flow - ole

#### iOS user-facing changes

-

#### Android user-facing changes

-

#### Dev changes

-

</details>

[CX-1419]: https://artsyproduct.atlassian.net/browse/CX-1419

### CODE REVIEW

**Comment by ArtsyOpenSource** [comment]
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true">Warnings</th>
    </tr>
  </thead>
  <tbody><tr>
      <td>:warning:</td>
      <td>

  ❌ **An error occurred while validating your changelog, please make sure you provided a valid changelog.**
  </td>
    </tr>
  </tbody>
</table>



<p align="right">
  Generated by :no_entry_sign: <a href="https://danger.systems/js">dangerJS</a> against 918a55e13b7f997d22ab24bc1dbac34efb63dd60
</p>

File: package.json [hunk 1]
```diff
@@ -56,7 +56,7 @@
     "storybook-server": "start-storybook -p 7007",
     "build-storybook": "build-storybook",
     "start-storybook": "STORYBOOK=1 yarn start",
-    "detect-secrets": "detect-secrets-launcher",
+    "detect-secrets": "detect-secrets-launcher --baseline .secrets.baseline",
```
Comment (olerichter00) [inline package.json:59]: **Note:** Just cleaning up here.

File: src/lib/Scenes/Onboarding/OnboardingCreateAccount/OnboardingCreateAccount.tsx [hunk 2]
```diff
@@ -244,41 +250,10 @@ export const OnboardingCreateAccountButton: React.FC<OnboardingCreateAccountButt
...
-              >
-                Conditions of Sale
-              </Text>
-              .
-            </Text>
-          </Flex>
-        </Touchable>
+        <Flex width="100%" pr={3} my={2}>
+          <TermsOfServiceCheckbox setChecked={setAcceptedTerms} checked={acceptedTerms} error={highlightTerms} />
+          <EmailSubscriptionCheckbox setChecked={setAgreedToReceiveEmails} checked={agreedToReceiveEmails} />
```
Comment (olerichter00) [inline src/lib/Scenes/Onboarding/OnboardingCreateAccount/OnboardingCreateAccount.tsx:255]: **Note:** I separated both checkboxes.

File: src/lib/store/AuthModel.ts [hunk 3]
```diff
@@ -92,14 +96,28 @@ export interface AuthModel {
...
+        agreedToReceiveEmails: boolean
       },
     {},
     GlobalStoreModel,
     Promise<boolean>
   >
-  authFacebook: Thunk<AuthModel, { signInOrUp: "signIn" | "signUp" }, {}, GlobalStoreModel, Promise<true>>
-  authGoogle: Thunk<AuthModel, { signInOrUp: "signIn" | "signUp" }, {}, GlobalStoreModel, Promise<true>>
-  authApple: Thunk<AuthModel, undefined, {}, GlobalStoreModel, Promise<true>>
+  authFacebook: Thunk<
```
Comment (olerichter00) [inline src/lib/store/AuthModel.ts:106]: **Note:** I added the flag to all signup actions (normal, Google, Facebook, Apple).

File: package.json [hunk 4]
```diff
@@ -277,7 +277,7 @@
       "yarn prettier-write"
     ],
     "**/!(package.json)": [
-      "detect-secrets-launcher --baseline .secrets.baseline"
+      "yarn detect-secrets"
```
Comment (brainbicycle) [inline package.json:280]: 🧹

**Review by brainbicycle — APPROVED** [review]
🎉

**Pull request merged**